src/robot/model/testcase.py

```diff
@@ -189,14 +198,16 @@ def to_dict(self):
         return data
 
 
-class TestCases(ItemList):
+class TestCases(ItemList[TestCase]):
     __slots__ = []
 
-    def __init__(self, test_class=TestCase, parent=None, tests=None):
+    def __init__(self, test_class: Type[TestCase]= TestCase,
+                 parent: 'TestSuite|None' = None, tests: Sequence[TestCase] = ()):
```

Comment:
Adding Mapping here raises an error from vscode. Looks like `ItemList` needs to be updated to use the newly agreed-upon conventions (stringified type hints, using mapping instead of dict, etc).